dual_arm_teleop/src/robot_arm.cpp

```diff
@@ -0,0 +1,297 @@
+
+  relative_pose.translate(translation);
+
+  // Rotation
+  relative_pose.rotate(viveToVictorRotation(quatMsgToEigen(msg_controller.posestamped.pose.orientation)));
+  relative_pose.rotate(viveToVictorRotation(controller_start_rotation).inverse());
+
+  // Generate IK solutions
+  const kinematics::KinematicsBaseConstPtr& solver = joint_model_group->getSolverInstance();
+  assert(solver.get());
+
+  Eigen::Affine3d solverTrobot = Eigen::Affine3d::Identity();
+  kinematic_state->setToIKSolverFrame(solverTrobot, solver);
+
+  // Convert to solver frame
+  Eigen::Affine3d pt_solver = solverTrobot * relative_pose;
+
+  std::vector<geometry_msgs::Pose> target_poses;
+  geometry_msgs::Pose pose;
+  Eigen::Quaterniond q(pt_solver.linear());
+  pose.position.x = pt_solver.translation().x();
+  pose.position.y = pt_solver.translation().y();
+  pose.position.z = pt_solver.translation().z();
+  pose.orientation.x = q.x();
+  pose.orientation.y = q.y();
+  pose.orientation.z = q.z();
+  pose.orientation.w = q.w();
+  target_poses.push_back(pose);
+
+  std::vector<double> seed = joint_position_measured;
+  std::vector<std::vector<double>> solutions;
+  kinematics::KinematicsResult result;
+  kinematics::KinematicsQueryOptions options;
+  options.discretization_method = kinematics::DiscretizationMethod::ALL_DISCRETIZED;
+
+  solver->getPositionIK(target_poses, seed, solutions, result, options);
+
+  // Pick the solution that matches closest to the measured joint state
+  if (!solutions.empty()) {
+    SeedDistanceFunctor functor(seed);
+    std::priority_queue<std::vector<double>, std::vector<std::vector<double>>, SeedDistanceFunctor> slnQueue(solutions.begin(), solutions.end(), functor);
+    kinematic_state->setJointGroupPositions(joint_model_group, slnQueue.top());
+
+    last_valid_pose = relative_pose;
+    ee_start_translation = relative_pose.translation();
+
+    err_msg.text = "";
+  }
+  else
+  {
+    err_msg.text = "Did not find IK solution";
+  }
+
+  std::cerr << "Got " << solutions.size() << " solutions for " << joint_model_group->getName() << std::endl;
+
+  // Arm control
+  victor_hardware_interface::MotionCommand msg_out_arm;
+  msg_out_arm.control_mode.mode = 2;
+
+  std::vector<double> joint_values;
+  kinematic_state->copyJointGroupPositions(joint_model_group, joint_values);
+
+  msg_out_arm.joint_position.joint_1 = joint_values[0];
+  msg_out_arm.joint_position.joint_2 = joint_values[1];
+  msg_out_arm.joint_position.joint_3 = joint_values[2];
+  msg_out_arm.joint_position.joint_4 = joint_values[3];
+  msg_out_arm.joint_position.joint_5 = joint_values[4];
+  msg_out_arm.joint_position.joint_6 = joint_values[5];
+  msg_out_arm.joint_position.joint_7 = joint_values[6];
+
+  controller_start_translation = pointMsgToEigen(msg_controller.posestamped.pose.position);
+
+  // Gripper control
+  victor_hardware_interface::Robotiq3FingerCommand msg_out_gripper;
+
+  victor_hardware_interface::Robotiq3FingerActuatorCommand scissor;
```

Comment:
Make a function to initialize all these value to avoid cluttering the main control method.
This function belongs in arc_utilities